src/main/java/org/jabref/logic/pdf/search/indexing/DocumentReader.java

```diff
@@ -0,0 +1,94 @@
+package org.jabref.logic.pdf.search.indexing;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.document.TextField;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.util.PDFTextStripper;
+
+import static org.jabref.model.pdf.search.SearchFieldConstants.AUTHOR;
+import static org.jabref.model.pdf.search.SearchFieldConstants.CONTENT;
+import static org.jabref.model.pdf.search.SearchFieldConstants.CREATOR;
+import static org.jabref.model.pdf.search.SearchFieldConstants.KEY;
+import static org.jabref.model.pdf.search.SearchFieldConstants.KEYWORDS;
+import static org.jabref.model.pdf.search.SearchFieldConstants.SUBJECT;
+
+public final class DocumentReader {
+
+    private final BibEntry entry;
+    private final PDFTextStripper pdfTextStripper = new PDFTextStripper();
+
+    public DocumentReader(BibEntry bibEntry) throws IOException {
+        if (!bibEntry.getField(FieldName.FILE).isPresent()) {
+            throw new IllegalArgumentException("The file field must not be absent when trying to reading the document!");
+        }
+
+        this.entry = bibEntry;
+        pdfTextStripper.setLineSeparator("\n");
+    }
+
+    /**
+     * Reads the content and metadata from a pdf file
+     */
+    public Document readPdfContents() throws IOException {
+        Path pdfPath = Paths.get(this.entry.getField(FieldName.FILE).get());
```

Comment:
The format of the files field is more complicated than a simple file path. In the end it contains a list of [linked files](https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/entry/LinkedFile.java). You can get the list of linked files using `BibEntry.getFiles()`.